### PR TITLE
Fix:  Few bugs during plan run/apply

### PR DESCRIPTION
**Pull request by mykalmax**

### CODE REVIEW

File: web/client/src/library/components/plan/Plan.tsx [hunk 1]
```diff
@@ -58,6 +62,10 @@ function Plan({
           isFalse(planOverviewTracker.isBackfillUpdate)))
     ) {
       setPlanAction(EnumPlanAction.Done)
+    } else if (isFetchingPlan) {
+      setPlanAction(EnumPlanAction.Running)
+    } else if (isFetchingPlanApply) {
+      setPlanAction(EnumPlanAction.Applying)
```
Comment (mykalmax) [inline web/client/src/library/components/plan/Plan.tsx:68]: this entire block  needs refactor, will do it in separate PR

File: web/client/src/library/components/editor/extensions/help.ts [hunk 2]
```diff
@@ -127,8 +127,11 @@ function getMarkDecorations(
 
     visisted.add(key)
 
-    const regex = new RegExp(name, 'ig')
-    const regex_normalized = new RegExp(alternativeNameFormat(name), 'ig')
+    const regex = new RegExp(`\\b${name}\\b`, 'ig')
```
Comment (mykalmax) [inline web/client/src/library/components/editor/extensions/help.ts:130]: name should not be a `substring`

File: web/client/src/library/components/progress/Progress.tsx [hunk 3]
```diff
@@ -1,17 +1,25 @@
+import { isFalse } from '@utils/index'
 import clsx from 'clsx'
 
-export interface PropsProgress extends React.HTMLAttributes<HTMLElement> {
-  progress: number
-  delay?: number
-  duration?: number
-}
+const PROGRESS_START = 3
```
Comment (vchan) [inline web/client/src/library/components/progress/Progress.tsx:4]: What does this 3 mean?

**Review by vchan — APPROVED** [review]

File: web/client/src/library/components/progress/Progress.tsx (hunk 3 above)
Comment (mykalmax) [reply web/client/src/library/components/progress/Progress.tsx:4]: In some cases we want to start progress not from 0 to indicate that process started but no progress has been recorded yet. And `PROGRESS_START` sets starting point. 
Screenshot 2023-11-28 at 10 07 05 AM
